document/src/main/java/com/yahoo/document/update/TensorModifyUpdate.java

```diff
@@ -0,0 +1,119 @@
+    protected TensorFieldValue tensor;
+
+    public TensorModifyUpdate(Operation operation, TensorFieldValue tensor) {
+        super(ValueUpdateClassID.TENSORMODIFY);
+        this.operation = operation;
+        this.tensor = tensor;
+    }
+
+
+    public Operation getOperation() { return operation; }
+    public TensorFieldValue getTensor() { return tensor; }
+
+    /** Returns the tensor */
+    public TensorFieldValue getValue() { return tensor; }
+
+    /** Sets the tensor */
+    public void setValue(TensorFieldValue value) { tensor = value; }
+
+
+    @Override
+    public FieldValue applyTo(FieldValue oldValue) {
+        if (oldValue instanceof TensorFieldValue) {
+        } else {
+            throw new IllegalStateException("Cannot use tensor modify update on non-tensor datatype "+oldValue.getClass().getName());
+        }
+        return oldValue;
+    }
+
+    @Override
+    protected void checkCompatibility(DataType fieldType) {
+        if (!(fieldType instanceof TensorDataType)) {
+            throw new UnsupportedOperationException("Expected tensor type, got " + fieldType.getName() + ".");
+        }
+    }
+
+    @Override
+    public void serialize(DocumentUpdateWriter data, DataType superType) {
+        data.write(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        return o instanceof TensorModifyUpdate && super.equals(o) &&
+                operation == ((TensorModifyUpdate) o).operation && tensor.equals(((TensorModifyUpdate) o).tensor);
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode() + operation.hashCode() + tensor.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return super.toString() + " " + operation.name + " " + tensor;
+    }
+
+    /**
+     * Lists valid operations that can be performed by a TensorModifyUpdate.
+     */
+    public enum Operation {
+        /**
+         * Replace tensor cell values with values from matching update tensor cells.
+         */
+        REPLACE(0, "replace"),
+        /**
+         * Add values from matching update tensor cells to target tensor cells.
+         */
+        ADD(1, "add"),
+        /**
+         * Multiply values from matching update tensor cells with target tensor cells.
+         */
+        MUL(2, "multiply");
+
+        /**
+         * The numeric ID of the operator, used for serialization.
+         */
+        public final int id;
+        /**
+         * The name of the operator, mainly used in toString() methods.
+         */
+        public final String name;
+
+        Operation(int id, String name) {
+            this.id = id;
+            this.name = name;
+        }
+
+        /**
+         * Returns the operator with the specified ID.
+         *
+         * @param id the ID to search for
+         * @return the Operator with the specified ID, or null if it does not exist.
+         */
+        public static Operation getID(int id) {
```

Comment:
Consider renaming to getOperation(). Also I think we can delete the function comment as it doesn't provide any extra value.